src/common/viz/FigureExtractor.js

```diff
@@ -17,210 +17,294 @@ define(['./Utils'], function (Utils) {
+                title: nodeInfo.attributes.title
+            };
 
-    FigureExtractor.prototype[EXTRACTORS.PLOT2D] = function (node) {
-        let desc = this[EXTRACTORS.SUBGRAPH](node);
-        desc.images = this.extractChildrenOfType(node, EXTRACTORS.IMAGE);
-        return desc;
-    };
+            desc.subGraphs = nodeInfo.children.map((childInfo) => {
+                const childNodeFn = this.getMetaType(childInfo);
+                ensureCanExtract(childNodeFn);
+                return this[childNodeFn](childInfo);
+            });
 
-    FigureExtractor.prototype[EXTRACTORS.PLOT3D] = function(node) {
-        let desc = this[EXTRACTORS.SUBGRAPH](node);
-        desc.zlim = node.getAttribute('zlim');
-        desc.zlabel = node.getAttribute('zlabel');
-        return desc;
-    };
+            return desc;
+        }
 
-    FigureExtractor.prototype[EXTRACTORS.LINE] = function (node) {
-        const id = node.getId(),
-            execId = this.getExecutionId(node);
-        let points, desc;
-
-        points = node.getAttribute('points').split(';')
-            .filter(data => !!data)  // remove any ''
-            .map(pair => extractPointsArray(pair));
-
-        desc = {
-            id: id,
-            execId: execId,
-            subgraphId: this._client.getNode(node.getParentId()).getAttribute('id'),
-            lineName: node.getAttribute('name'),
-            label: node.getAttribute('label'),
-            lineWidth: node.getAttribute('lineWidth'),
-            marker: node.getAttribute('marker'),
-            name: node.getAttribute('name'),
-            type: 'line',
-            points: points,
-            color: node.getAttribute('color')
-        };
-        return desc;
-    };
+        [EXTRACTORS.SUBGRAPH] (nodeInfo) {
+            const id = nodeInfo.id,
+                graphId = nodeInfo.parent.id,
+                execId = this.getExecutionId(nodeInfo);
+            let desc;
 
-    FigureExtractor.prototype[EXTRACTORS.IMAGE] = function (node) {
-        const id = node.getId(),
-            execId = this.getExecutionId(node),
-            imageHeight = node.getAttribute('height'),
-            imageWidth = node.getAttribute('width'),
-            numChannels = node.getAttribute('numChannels');
-        const colorModel = numChannels === 3 ? 'rgb' : 'rgba';
-        return {
-            id: id,
-            execId: execId,
-            subgraphId: this._client.getNode(node.getParentId()).getAttribute('id'),
-            type: 'image',
-            width: imageWidth,
-            height: imageHeight,
-            colorModel: colorModel,
-            visible: node.getAttribute('visible'),
-            rgbaMatrix: Utils.base64ToImageArray(node.getAttribute('rgbaMatrix'), imageWidth, imageHeight, numChannels)
-        };
-    };
+            desc = {
+                id: id,
+                execId: execId,
+                type: this.getMetaType(nodeInfo) === EXTRACTORS.PLOT3D ? 'plot3D' : 'plot2D',
```

Comment:
FigureExtractor `desc.type` is lowercase  `Graph` nodes and below. It might be nice to change it to use upper case. However, it might not be a priority considering #1745 / #1747 will not use this code.